model-optimizer/mo/front/kaldi/loader/loader.py

```diff
@@ -44,52 +43,52 @@ def load_parallel_component(file_descr, graph: Graph, prev_layer_id):
     nnet_count = read_token_value(file_descr, b'<NestedNnetCount>')
     log.debug('Model contains parallel component with {} nested networks'.format(nnet_count))
 
-    slice_id = graph.unique_id(prefix='Slice')
-    graph.add_node(slice_id, parameters=None, op='slice', kind='op')
-
-    slice_node = Node(graph, slice_id)
-    Node(graph, prev_layer_id).add_output_port(0)
-    slice_node.add_input_port(0)
-    graph.create_edge(Node(graph, prev_layer_id), slice_node, 0, 0)
-    slices_points = []
-
+    split_points = []
     outputs = []
+    inputs = []
 
     for i in range(nnet_count):
         read_token_value(file_descr, b'<NestedNnet>')
         collect_until_token(file_descr, b'<Nnet>')
         g = Graph()
         load_kalid_nnet1_model(g, file_descr, 'Nested_net_{}'.format(i))
-        input_nodes = [n for n in graph.nodes(data=True) if n[1]['op'] == 'Parameter']
-        shape = input_nodes[0][1]['shape']
-        if i != nnet_count - 1:
-            slices_points.append(shape[1])
-        g.remove_node(input_nodes[0][0])
+
+        input_nodes = [n for n in g.nodes(data=True) if n[1]['op'] == 'Parameter']
+        if len(input_nodes) == 1:
+            shape = input_nodes[0][1]['shape']
+            split_points.append(shape[1])
+            g.remove_node(input_nodes[0][0])
+        else:
+            raise Error('NestedNnet with multiple inputs is not supported')
+
         mapping = {node: graph.unique_id(node) for node in g.nodes(data=False) if node in graph}
         g = nx.relabel_nodes(g, mapping)
         for val in mapping.values():
             g.node[val]['name'] = val
         graph.add_nodes_from(g.nodes(data=True))
         graph.add_edges_from(g.edges(data=True))
         sorted_nodes = tuple(nx.topological_sort(g))
-        edge_attrs = create_edge_attrs(slice_id, sorted_nodes[0])
-        edge_attrs['out'] = i
-        Node(graph, slice_id).add_output_port(i)
-        Node(graph, sorted_nodes[0]).add_input_port(len(Node(graph, sorted_nodes[0]).in_ports()))
-        graph.create_edge(Node(graph, slice_id), Node(graph, sorted_nodes[0]), i, 0)
+
         outputs.append(sorted_nodes[-1])
-    packed_sp = struct.pack("B", 4) + struct.pack("I", len(slices_points))
-    for i in slices_points:
-        packed_sp += struct.pack("I", i)
-    slice_node.parameters = io.BytesIO(packed_sp)
+        inputs.append(sorted_nodes[0])
+
+    split_id = graph.unique_id(prefix='NestedNets/Split')
+    mapping_rule = {'out_ports_count': nnet_count, 'size_splits': split_points, 'axis': 1, 'name': split_id}
```

Comment:
nit: please, rename the variable `mapping_rule` to `attrs` or similar

Reply:
done